src/prefect/engine/task_runner.py

```diff
@@ -901,28 +901,28 @@ def get_task_run_state(
         except signals.LOOP as exc:
             new_state = exc.state
             assert isinstance(new_state, Looped)
-            new_state.result = Result(
-                value=new_state.result, result_handler=self.result_handler
+            new_state.result = Result( # todo: here
+                value=new_state.result, result_handler=self.result_handler # todo: here
             )
             new_state.cached_inputs = inputs
             new_state.message = exc.state.message or "Task is looping ({})".format(
                 new_state.loop_count
             )
             return new_state
 
-        result = Result(value=result, result_handler=self.result_handler)
+        result = Result(value=result, result_handler=self.result_handler) # todo: here
```

Comment:
This is where I mutated the task's result a la `self.task.result.value=result` instead of creating a new Result object for them.

Reply:
So this sounds like we might want to use:

```python
result = self.task.result.format(**prefect.context)
result = result.write(value)
```

Which I think we should simplify to a single interface:
```python
result = result.write(value, **prefect.context)
```